sources/packages/backend/apps/workers/src/controllers/application/application.controller.ts

```diff
@@ -0,0 +1,114 @@
+import { Controller } from "@nestjs/common";
+import { ZeebeWorker } from "../../zeebe";
+import {
+  ZeebeJob,
+  MustReturnJobActionAcknowledgement,
+  IOutputVariables,
+  ICustomHeaders,
+} from "zeebe-node";
+import {
+  ApplicationExceptionService,
+  ApplicationService,
+} from "../../services";
+import {
+  ApplicationExceptionsJobInDTO,
+  ApplicationExceptionsJobOutDTO,
+  ApplicationUpdateStatusJobHeaderDTO,
+  ApplicationUpdateStatusJobInDTO,
+} from "..";
+import { APPLICATION_ID } from "../workflow-variables";
+import { ApplicationExceptionStatus } from "@sims/sims-db";
+import {
+  APPLICATION_NOT_FOUND,
+  APPLICATION_STATUS_NOT_UPDATED,
+} from "../error-code-constants";
+
+@Controller()
+export class ApplicationController {
+  constructor(
+    private readonly applicationService: ApplicationService,
+    private readonly applicationExceptionService: ApplicationExceptionService,
+  ) {}
+
+  /**
+   * Updates the student application status ensuring that the application
+   * was in the expected state and also allowing the method to be called
+   * multiple times without causing any harm to enure the impotency.
```

Comment:
did you mean "ensure the idempotency" here ?

Reply:
Fixed.